io/disk/sysfs_att.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Maram Srimannarayana Murthy <Maram.Srimannarayana.Murthy@ibm.com>
+
+"""
+Check all sysfs queue attributes by reading or writing different values
+"""
+
+import os
+from avocado import Test
+from avocado.utils import genio
+
+
+class Sysfs_Queue(Test):
+
+    """
+    Checks all Sysfs_Queue attributes
+    """
+
+    def setUp(self):
+        """
+        Reading required arguments from yaml file
+        Setting up file paths
+        """
+        self.queue_attr_disk = self.params.get("disk", default=None)
+        self.log.info("disk: %s", self.queue_attr_disk)
+        self.queue_attr_disk_dir = os.path.join("/sys/block",
+                                                self.queue_attr_disk, "queue")
+        self.attribute_name = self.params.get("attribute_name",
+                                              default="None")
+        self.attribute_values = self.params.get("values").split(' ')
+        if not self.queue_attr_disk:
+            self.cancel("No disk input, please update yaml and re-run")
+
+    def test(self):
+        """
+        Read sysfs queue readable attributes
+        Test sysfs queue attributes with different values
+        check functionality with different values
+        """
+        self.log.info("Reading attributes values from sysfs files")
+        for each_file in os.listdir(self.queue_attr_disk_dir):
+            each_file = os.path.join(self.queue_attr_disk_dir, each_file)
+            if os.access(each_file, os.R_OK):
+                self.log.info(f"Reading from file {each_file}")
+                self.log.info(genio.read_file(each_file).rstrip("\n"))
+        attribute_file = os.path.join(self.queue_attr_disk_dir,
+                                      self.attribute_name)
+        self.log.info(f"Setting up value to attribute file {attribute_file}")
+        for attribute_val in self.attribute_values:
+            genio.write_file(attribute_file, str(attribute_val))
+            attribute_file_val = genio.read_file(attribute_file).rstrip("\n")
+            if attribute_val != attribute_file_val:
+                self.fail(f"{attribute_file} file updation with value"
+                          f"{attribute_val} is failed")
```

Comment:
@maramsmurthy as discussed earlier, validating attribute values with other tools output will be a good scenario and please take care of time related parameter.

Reply:
Will be adding as suggested tin upcoming PR's. Meanwhile this can be used.